src/clib/pioc_support.cpp

```diff
@@ -3181,6 +3182,21 @@ int spio_createfile_int(int iosysid, int *ncidp, const int *iotype, const char *
         return check_mpi(ios, file, mpierr, __FILE__, __LINE__);
     }
 
+    if ((file->iotype == PIO_IOTYPE_ADIOS) || (file->iotype == PIO_IOTYPE_ADIOSC))
+    {
+#ifdef SPIO_NO_CXX_REGEX
+        LOG((1, "WARNING: Since C++11 regex support is not available, overriding ADIOS type with PnetCDF for file creation is skipped"));
+#else /* SPIO_NO_CXX_REGEX */
+        std::regex rgx(PIO_OVERRIDE_ADIOS_WITH_PNETCDF_REGEX);
+        if (std::regex_match(filename, rgx))
+        {
+#ifdef _PNETCDF
+            file->iotype = PIO_IOTYPE_PNETCDF;
```

Comment:
Can you add a warning that we are overriding the iotype?

Reply:
A warning has been added.